docs.md

```diff
@@ -0,0 +1,84 @@
+---
+name: Release Helper
+icon: https://woodpecker-ci.org/img/logo.svg
+
+description: Plugin for semi-automated releases.
+authors: Woodpecker Authors
+tags: [git, release]
+containerImage: woodpeckerci/plugin-ready-release-go
+containerImageUrl: https://hub.docker.com/r/woodpeckerci/plugin-ready-release-go
+url: https://github.com/woodpecker-ci/plugin-ready-release-go
+---
+
+# Introduction
+
+This plugin aims to help with git-based releases.
+It should be run on every commit of the default branch to execute it's necessary actions.
+
+A Woodpecker workflow file could look like this:
+
+```yaml
+steps:
+  release-helper:
+    image: woodpeckerci/plugin-ready-release-go:0.6.1
+    settings:
+      release_branch: ${CI_REPO_DEFAULT_BRANCH}
+      forge_type: github
+      git_email: <email>
+      github_token:
+        from_secret: GITHUB_TOKEN
+
+when:
+  event: push
+  branch: ${CI_REPO_DEFAULT_BRANCH}
+```
+
+## Features
+
+- Create automated changelog based on PRs which updates itself after each merge to the default branch
+- Auto-categorization of PRs based on labels
+- Automatically determines the next version
+- Supports any kind of programming language, changelog tool and commit style
+- Execute custom pre-release hooks
+
+## Overriding Settings
+
+To override the default settings, add a `release-config.ts` file to your repository.
```

Comment:
For the settings we might just point the user to the config in the source-code and write examples as comments.

Reply:
I think WRT examples there is not much more needed than what I added here - showcasing how to write `release-config.ts`. Users will be able to fill it accordingly with the settings values. 

I am not sure that putting more examples into the source code might help or is actually a good idea?

Reply:
Another Q here is: do you plan to make the repo "plugin-like"? As currently, it doesn't follow the "plugin structure" and reads the config from a file. But maybe it would be good to align the usage and allow both ways, the current file-based one and the `settings:` notation?